cmd/api/src/model/appcfg/parameter.go

```diff
@@ -56,26 +61,71 @@ type Parameter struct {
 
 // Map is a convenience function for mapping the data stored in the Value Parameter struct member onto
 // a richer type provided by the given value.
-func (s Parameter) Map(value any) error {
+func (s *Parameter) Map(value any) error {
 	return s.Value.Map(value)
 }
 
-func (s Parameter) IsValid(parameter string) bool {
-	availParams, err := AvailableParameters()
-	if err != nil {
-		log.Errorf("Error occurred getting AvailableParamters: %v", err)
-		return false
+func (s *Parameter) IsValidKey(parameterKey string) bool {
+	validKeys := map[string]bool{
+		PasswordExpirationWindow: true,
+		Neo4jConfigs:             true,
+		PruneTTL:                 true,
+		CitrixRDPSupportKey:      true,
+		ReconciliationKey:        true,
+	}
+
+	return validKeys[parameterKey]
+}
+
+func (s *Parameter) Validate() utils.Errors {
+	// validate the base parameter
+	var (
+		objMap map[string]any
+		ok     bool
+	)
+	if objMap, ok = s.Value.Object.(map[string]any); !ok || len(objMap) == 0 {
+		return utils.Errors{errors.New("missing or invalid property: value")}
+	}
+
+	// validate the specific parameter value
+	var v any
+	switch s.Key {
+	case PasswordExpirationWindow:
+		v = &PasswordExpiration{}
+	case Neo4jConfigs:
+		v = &Neo4jParameters{}
+	case PruneTTL:
+		v = &PruneTTLParameters{}
+	case CitrixRDPSupportKey:
+		v = &CitrixRDPSupport{}
+	case ReconciliationKey:
+		v = &ReconciliationParameter{}
+	default:
+		return utils.Errors{errors.New("invalid key")}
+	}
+
+	// numField panics when val is not a struct, so we need both checks
+	if val := reflect.Indirect(reflect.ValueOf(v)); val.Kind() != reflect.Struct || val.NumField() != len(objMap) {
+		return utils.Errors{errors.New("value property contains an invalid field")}
```

Comment:
Not a huge fan of reflect use here but didn't know of a better way to count the fields and ensure no added fields were included.

Reply:
Not a golang expert by any means, but I think an approach like this might be hard to avoid given the way this endpoint currently works. We have a similar issue on the UI side where we are having to do some runtime matching to determine the correct data type per config.